src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -1816,6 +1816,46 @@ def test_etc_ceph_init(self):
         with with_cephadm_module({'manage_etc_ceph_ceph_conf': True}) as m:
             assert m.manage_etc_ceph_ceph_conf is True
 
+    @mock.patch("cephadm.CephadmOrchestrator.check_mon_command")
+    @mock.patch("cephadm.CephadmOrchestrator.extra_ceph_conf")
+    def test_extra_ceph_conf(self, _extra_ceph_conf, _check_mon_cmd, cephadm_module: CephadmOrchestrator):
+        # settings put into the [global] section in the extra conf
+        # need to be appended to existing [global] section in given
+        # minimal ceph conf, but anything in another section (e.g. [mon])
+        # needs to continue to be its own section
+
+        # this is the conf "ceph generate-minimal-conf" will return in this test
+        _check_mon_cmd.return_value = (0, """[global]
+global_k1 = global_v1
+global_k2 = global_v2
+[mon]
+mon_k1 = mon_v1
+[osd]
+osd_k1 = osd_v1
+osd_k2 = osd_v2
+""", '')
+
+        # test with extra ceph conf that has some of the sections from minimal conf
+        _extra_ceph_conf.return_value = CephadmOrchestrator.ExtraCephConf(conf="""[mon]
+mon_k2 = mon_v2
+[global]
+global_k3 = global_v3
+""", last_modified=datetime_now())
+
+        expected_combined_conf = """[global]
+global_k1 = global_v1
+global_k2 = global_v2
+global_k3 = global_v3
+[mon]
+mon_k1 = mon_v1
+mon_k2 = mon_v2
+[osd]
+osd_k1 = osd_v1
+osd_k2 = osd_v2
+"""
```

Comment:
forgot to check make check failure before leacing my initial review. @Svelar it seems the code here is working very slightly differently than mine and causing the unit test to fail.
```
E           AssertionError: assert '[global]\ngl...k2 = osd_v2\n' == '[global]\ngl...k2 = osd_v2\n'
E             Skipping 65 identical leading characters in diff, use -v to show
E               global_v3
E             + 
E               [mon]
E               mon_k1 = mon_v1
E               mon_k2 = mon_v2
E             + ...
E             
E             ...Full output truncated (4 lines hidden), use '-vv' to show
```
I tested locally and it passes if you just add a blank line at the end of each section in the expected conf string in the test (and that blank line should be fine in the conf, so we should be okay just updating the test)
```
bash-5.2$ git diff
diff --git a/src/pybind/mgr/cephadm/tests/test_cephadm.py b/src/pybind/mgr/cephadm/tests/test_cephadm.py
index 663ef522ff4..5424f210922 100644
--- a/src/pybind/mgr/cephadm/tests/test_cephadm.py
+++ b/src/pybind/mgr/cephadm/tests/test_cephadm.py
@@ -1846,9 +1846,11 @@ global_k3 = global_v3
 global_k1 = global_v1
 global_k2 = global_v2
 global_k3 = global_v3
+
 [mon]
 mon_k1 = mon_v1
 mon_k2 = mon_v2
+
 [osd]
 osd_k1 = osd_v1
 osd_k2 = osd_v2
```

Reply:
Thanks for pointing out, I also forgot that. I have modified the code, and then passed all checks.